pkgs/top-level/all-packages.nix

```diff
@@ -12212,7 +12212,7 @@ in
 
   dbxml = callPackage ../development/libraries/dbxml { };
 
-  dbus = callPackage ../development/libraries/dbus { };
+  dbus = callPackage ../development/libraries/dbus { systemd = systemdMinimal; };
```

Comment:
As noted on an outdated part of the PR (sorry for that),

Instead of depending `systemdMinimal` lets just remove the `dbus` dependency on our toplevel `systemd` package. As it's only used in the test suite (which afaik we disable anyway).    Then we do not need two copies of systemd in our closure.  

(Though we still have to keep it for `udev` I guess; but that's a different issue, not related to this PR)

Reply:
I'm pretty sure systemd-homed needs dbus

Reply:
That would really surprise me. But I'm happy to be proven wrong.   All systemd services usually use the built-in `sd-bus` library for dbus support, for bootstrapping reasons.

Reply:
Also searching through the source code; only a test seems to import `dbus/dbus.h` :

https://github.com/systemd/systemd/search?q=%22dbus%2Fdbus.h%22&type

So homed requiring `libdbus` seems very unlikely to me..

Reply:
You seem to be right. I had some problems compiling homed in the beginning, and they did go away as I added dbus and some other libraries. Will see what happens without dbus (I think it should work).

Reply:
removed dbus as input from systemd. do we still want to keep systemdMinimal?

Reply:
I'd say no. As it's not really related to your PR anymore. It cleans up the diff a bit.

Reply:
I'd prefer to keep it - due to https://github.com/NixOS/nixpkgs/pull/98299#discussion_r495240873 and as a alternative to https://github.com/NixOS/nixpkgs/pull/97051.

Reply:
I also used it to compile libfido2